Rework type option explanation (#119)
bufbuild/docs.buf.build: docs/build/usage.md

Versions 1.1.0 and later of the `buf` CLI include a `--type` option for the `buf build` command that
enables you to supply a fully qualified Protobuf name and limit the resulting image or
`FileDescriptorSet` to only those descriptors required to represent those types and their required
dependencies. This example usage restricts the output types to those required to represent
`pkg.foo.Bar`:

```terminal
$ buf build --type pkg.foo.Bar
```

### Supported constructs {#constructs}

The `--type` option supports these Protobuf constructs:

- [Messages], including:
  - Messages and enums referenced in message fields
  - Any [proto2] extension declarations for message fields
  - The parent message if this message is a nested definition
  - Any custom options for the message, its fields, and the file in which the message is defined
- [Enums], including:
  - The enum value descriptors for this enum
  - The parent message is this enum is a nested definition
  - Any custom options used in the enum, enum values, and the file in which the enum is defined
- [Services], including:
  - Request and response types referenced in service methods

>   - Any custom options for the services, its methods, and the file in which the service is defined

Reviewer: Here the `including` parts don't really relay well the message I initially wanted to get across. The goal is to describe what counts as a "required dependency" in the part above:
>  only those descriptors required to represent those types and their required
dependencies

I think now it looks more like these sub bullets refer to the "including".

Author: @robbertvanginkel I'm still struggling a bit with the wording here, but I made another pass at this.

## Limit to specific types

When you run `buf build` to create a [`FileDescriptorSet`][filedescriptorset] or Buf [image], the
output contains all of the Protobuf types declared in the [module] by default. But for some advanced
use cases, you may want the image or `FileDescriptorSet` to contain only a subset of the types
described in your Protobuf API.

Versions 1.1.0 and later of the `buf` CLI include a `--type` option for the `buf build` command that
enables you to supply a fully qualified Protobuf name and limit the resulting image or
`FileDescriptorSet` to only those descriptors required to represent those types and their required
dependencies. This example usage restricts the output types to those required to represent
`pkg.foo.Bar`:

```terminal
$ buf build --type pkg.foo.Bar
```

> Three Protobuf constructs can act as descriptors: [messages], [enums], and [services]. Each of these
> constructs has a set of dependent descriptors that are included in the build:

Reviewer: ```suggestion
The type flag accepts fully qualified names for [messages], [enums], and [services]. The following qualify as dependent descriptors that are included in the build:
```

Reviewer: There are more than 3 descriptors, they're just not supported here as top level arguments. How about this?

- [Messages]
  - Messages and enums referenced in message fields
  - Any [proto2] extension declarations for message fields
  - The parent message if this message is a nested definition
  - Any custom options for the message, its fields, and the file in which the message is defined
- [Enums]
  - The enum value descriptors for this enum
  - The parent message is this enum is a nested definition

>   - Any custom options used in the enum, enum values, and the file in which the enum is defined

Reviewer: ```suggestion
  - Any custom options for the enum, enum values, and the file in which the enum is defined
```
consistency with the other 2.

- [Services]
  - Request and response types referenced in service methods
  - Any custom options for the services, its methods, and the file in which the service is defined

:::success Supplying multiple types
You can specify multiple types by applying the `--type` option multiple times, as in this example:

```terminal
$ buf build \
  --type acme.weather.v1.Units \
  --type acme.weather.v1.CurrentWeather.Temperature
```

> In this case, all dependent descriptors are built for both `acme.weather.v1.Units` and
> `acme.weather.v1.CurrentWeather.Temperature`.

Reviewer: ```suggestion
In this case, dependent descriptors for both `acme.weather.v1.Units` and
`acme.weather.v1.CurrentWeather.Temperature` are included in the output.
```
Nit: This feature will still build a whole module, it just restricts the output of the build.
